### PR TITLE
executor: fix flashback table faild in uppercase database name

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Fix below bugs:

```sql
drop database if exists Test2;
create database Test2;
use Test2;
create table t (a int);
drop table t;
flashback table t; -- should execute successfully.
(1105, u"Can't find dropped/truncated table: t in DDL history jobs")

drop table t;
drop database if exists Test3;
create database Test3;
use Test3;
create table t (a int);
drop table t;
drop database Test3;
use Test2;
flashback table t; -- should execute successfully.
(1049, u"Unknown database '(Schema ID 62)'")
```

### What is changed and how it works?

No

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

Side effects

- No

### Release note 

- Fix flashback table failed in some special cases.

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -4843,6 +4843,23 @@ func (s *testRecoverTable) TestFlashbackTable(c *C) {
...
+
+	tk.MustExec("drop table t")
+	tk.MustExec("drop database if exists Test3")
+	tk.MustExec("create database Test3")
+	tk.MustExec("use Test3")
+	tk.MustExec("create table t (a int);")
+	tk.MustExec("drop table t")
+	tk.MustExec("drop database Test3")
+	tk.MustExec("use Test2")
+	tk.MustExec("flashback table t")
```
Comment (zimulala) [inline executor/executor_test.go:4862]: Do we need to check the existence of "t" and "Test3"?

Comment (crazycs520) [reply executor/executor_test.go:4864]: great, done.

File: executor/ddl.go [hunk 2]
```diff
@@ -518,9 +518,7 @@ func (e *DDLExec) getRecoverTableByTableName(tableName *ast.TableName) (*model.J
 		}
 		schema, ok := dom.InfoSchema().SchemaByID(job.SchemaID)
 		if !ok {
-			return true, infoschema.ErrDatabaseNotExists.GenWithStackByArgs(
-				fmt.Sprintf("(Schema ID %d)", job.SchemaID),
-			)
+			return false, nil
```
Comment (tangenta) [inline executor/ddl.go:521]: Why not check schema existence?

Comment (crazycs520) [reply executor/ddl.go:521]: because of the schema may be dropped, and the job is not the job that we need.

**Pull request merged**